src/plugins/data/public/search/create_app_mount_context_search.ts

```diff
@@ -48,7 +49,17 @@ export const createAppMountSearchContext = (
     const strategyPromise = getSearchStrategy(strategyName);
     return from(strategyPromise).pipe(
       mergeMap(strategy => {
-        return strategy.search(request, options);
+        loadingCount$.next(loadingCount$.getValue() + 1);
+        const search$ = strategy.search(request, options);
```

Comment:
I think actually calling `subscribe` on this observable is changing the behavior we'd like (as it now returns a hot observable rather than cold). I think something like this would be better and allow control to still be in the consumer's hands:

```ts
return strategy.search(request, options).pipe(
  tap(
    null,
    error => loadingCount$.next(loadingCount$.getValue() - 1),
    complete => loadingCount$.next(loadingCount$.getValue() - 1)
  )
);
```